packages/cli/src/workflows/workflow.service.ts

```diff
@@ -166,10 +166,7 @@ export class WorkflowService {
 		);
 
 		if (tagIds && !config.getEnv('workflowTagsDisabled')) {
-			await this.workflowTagMappingRepository.delete({ workflowId });
-			await this.workflowTagMappingRepository.insert(
-				tagIds.map((tagId) => ({ tagId, workflowId })),
-			);
+			await this.workflowTagMappingRepository.overwriteTaggings(workflowId, tagIds);
```

Comment:
Shouldn't we update the tags in the same transaction as we update the workflow itself?

Reply:
Related: https://n8nio.slack.com/archives/C069HS026UF/p1706783989128759?thread_ts=1706782548.306949&cid=C069HS026UF